<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>855</width>
    <height>603</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TextSecure by zhiyan114</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>RSA Key Management</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0" rowspan="2">
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Key Management</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0" colspan="3">
            <widget class="QPlainTextEdit" name="PublicKeyInput">
             <property name="placeholderText">
              <string>Your public RSA Key</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="3">
            <widget class="QPlainTextEdit" name="PrivateKeyInput">
             <property name="readOnly">
              <bool>false</bool>
             </property>
             <property name="placeholderText">
              <string>Your Private RSA Key</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="GenRSAKey">
             <property name="text">
              <string>Generate RSA Key</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="GenKeySize">
             <property name="inputMask">
              <string/>
             </property>
             <property name="text">
              <string>2048</string>
             </property>
             <property name="maxLength">
              <number>5</number>
             </property>
             <property name="placeholderText">
              <string>Key Size</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="PriToPubKeyBtn">
             <property name="text">
              <string>Extract Public Key</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>Private Key Encryption</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0" colspan="2">
            <widget class="QLineEdit" name="PriKeyPassInput">
             <property name="placeholderText">
              <string>RSA Private Key Passphrase</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="PriKeyEncBtn">
             <property name="text">
              <string>Encrypt</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="PriKeyDecBtn">
             <property name="text">
              <string>Decrypt</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QGroupBox" name="groupBox_7">
          <property name="title">
           <string>Certificate To Public Key Converter</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="0" column="0">
            <widget class="QPlainTextEdit" name="PublicCertInput">
             <property name="placeholderText">
              <string>Your Public RSA Certificate</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="ImportPubCert">
             <property name="text">
              <string>Import Certificate Key</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="AesSection">
       <attribute name="title">
        <string>AES Encryption</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="title">
           <string>Encryption Mode</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_8">
           <item row="0" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="EncInput">
             <property name="placeholderText">
              <string>Your Data Input Here</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="EncOutput">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>Your Data Input Here</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="EncryptBtn">
             <property name="text">
              <string>Encrypt</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="EncClear">
             <property name="text">
              <string>Clear Output</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Decryption Mode</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_9">
           <item row="4" column="0">
            <widget class="QPushButton" name="DecryptBtn">
             <property name="text">
              <string>Decrypt</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="DecOutput">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>Your Data Input Here</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="DecInput">
             <property name="placeholderText">
              <string>Your Data Input Here</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QLabel" name="MessageStatus">
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Message is original: NULL</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QPushButton" name="DecClear">
             <property name="text">
              <string>Clear Output</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLineEdit" name="KeyInput">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="inputMask">
             <string/>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="maxLength">
             <number>32</number>
            </property>
            <property name="placeholderText">
             <string>Your AES Key Here</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Preferred</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>10</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="isRawKey">
            <property name="text">
             <string>Use Raw Key</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="RSASection">
       <attribute name="title">
        <string>RSA Encryption</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_12">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Encryption</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_10">
           <item row="0" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="REncInput">
             <property name="placeholderText">
              <string>Your data input here</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="REncOutput">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>Your data output here</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="REncryptBtn">
             <property name="text">
              <string>Encrypt</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="REncryptClear">
             <property name="text">
              <string>Clear Output</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>Decryption</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_11">
           <item row="0" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="RDecInput">
             <property name="placeholderText">
              <string>Your data input here</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="RDecOutput">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>Your data output here</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="RDecryptBtn">
             <property name="text">
              <string>Decrypt</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="RDecryptClear">
             <property name="text">
              <string>Clear Output</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>AES/RSA Encryption</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_14">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_15">
          <property name="title">
           <string>Encryption</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_13">
           <item row="0" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="AREncInput">
             <property name="placeholderText">
              <string>Your data input here</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="AREncOutput">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>Your data output here</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="AREncryptBtn">
             <property name="text">
              <string>Encrypt</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="AREncryptClear">
             <property name="text">
              <string>Clear Output</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBox_16">
          <property name="title">
           <string>Decryption</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_15">
           <item row="0" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="ARDecInput">
             <property name="placeholderText">
              <string>Your data input here</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="ARDecOutput">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>Your data output here</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QLabel" name="ARMessageStatus">
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Message is original: NULL</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QPushButton" name="ARDecryptBtn">
             <property name="text">
              <string>Decrypt</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="ARDecryptClear">
             <property name="text">
              <string>Clear Output</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="UsageSection">
       <attribute name="title">
        <string>Software Usage/Information</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_17">
        <item row="0" column="0" colspan="2">
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <pointsize>20</pointsize>
            <bold>true</bold>
            <underline>true</underline>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color: rgb(0, 230, 230);</string>
          </property>
          <property name="text">
           <string>Software Usages/Information</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="1" column="0" rowspan="2" colspan="2">
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;You may use the software for any purposes and I will not be liable for the outcome. Both AES and RSA mode are available. It is recommended to use AES mode for message encryption then use the RSA as a AES Key encrytion since it has a severe size limit and is not designed for that purpose either way. (In a later release, a combination of both mode will be available so that you do not need to store the AES Key and data seperately).&lt;/p&gt;&lt;p&gt;The AES/RSA is simply a way to encrypt data by using an RSA key. It works by encrypting a random key and then embed it with the data which is encrypted with AES. If you're using an RSA keysize that are less than 592-bit a 128-bit AES key will be used instead of 256-bit AES key.&lt;/p&gt;&lt;p&gt;A software by &lt;span style=&quot; font-weight:600;&quot;&gt;zhiyan114&lt;/span&gt;. View the source code here: https://github.com/zhiyan114/SecureMessage&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="openExternalLinks">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>855</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
